docs/source/how-to/refresh.md

```diff
@@ -0,0 +1,132 @@
+This allows you to make sure things like group membership or other authorization info is up-to-date.
+In OAuth, this can also mean making sure the access token has not expired.
+This is particularly useful in deployments where an access token from the oauth provider is passed to the Server environment,
+e.g. for access to data sources, git repos, etc..
+You don't want to start a server passing an expired token, do you?
+
+OAuthenticator 17.2 introduces support in all OAuthenticator classes for refreshing user info via this mechanism, including requesting new access tokens if a `refresh_token` is available from the oauth provider.
+
+```{seealso}
+- [More about refresh tokens](https://www.oauth.com/oauth2-servers/making-authenticated-requests/refreshing-an-access-token/)
+```
+
+How it works:
+
+- Every time a user takes an authenticated action with JupyterHub
+  (making an API request, launching a server, visiting a page, etc.),
+  JupyterHub checks when the last time auth info was loaded from the provider.
+- If the auth info is older than [Authenticator.auth_refresh_age](inv:jupyterhub:py:attribute#jupyterhub.auth.Authenticator.auth_refresh_age), the auth info is refreshed,
+  i.e. the user model is retrieved anew with the current access token, and any changes are applied (usually there aren't any).
+  The default value for this age is five minutes.
+  You can consider it an expiring cache of the information we retrieved from the OAuth provider.
+- If the access token is expired and a refresh token is a available,
+  a new access token is retrieved via the [refresh_token grant](https://www.oauth.com/oauth2-servers/making-authenticated-requests/refreshing-an-access-token/)
+- If no auth info is retrievable (e.g. no refresh token and access token is expired or both are expired or revoked),
+  then the user must login again before they are able to take actions in JupyterHub
+  because at this point their authorization state is unknown and could no longer be valid.
+
+There is also an option [Authenticator.refresh_pre_spawn](inv:jupyterhub:py:attribute#jupyterhub.auth.Authenticator.refresh_pre_spawn) which can be enabled:
+
+```python
+c.Authenticator.refresh_pre_spawn = True
+```
+
+to ensure auth is up-to-date before launching a server.
+This is most useful when the server is being passed an access token
+because it ensures the token is valid when the server starts.
+
+## Refreshing tokens from user sessions
+
+```{warning}
+This example requires granting users read access to their own `auth_state`.
+If you plan to provide users with access tokens,
+`auth_state` does not typically include information your users won't have access to with the token itself,
+but it is worth making sure that your Authenticator configuration places anything in `auth_state`
```

Comment:
```suggestion
but it is worth making sure that your Authenticator configuration doesn't put anything in `auth_state`
```
Go ahead and merge after this fix (or equivalent)

Reply:
Thanks!